### PR TITLE
XW-3171 | banner notifications

### PR DESCRIPTION
**Pull request by latata**
https://wikia-inc.atlassian.net/browse/XW-3171
@Wikia/x-wing

### CODE REVIEW

File: extensions/wikia/BannerNotifications/js/BannerNotifications.js [hunk 1]
```diff
@@ -14,25 +14,57 @@
...
 	'use strict';
 
-	var defaultTimeout = 10000,
+	var templates = {
+			bannerNotification: '<div class="wds-banner-notification {{typeClassName}}">' +
+			'<div class="wds-banner-notification__icon">' +
+			'{{{icon}}}</div>' +
+			'<span class="wds-banner-notification__text">{{{content}}}</span>' +
+			// DS icon: wds-icons-cross-tiny
+			'<svg class="wds-icon wds-icon-tiny wds-banner-notification__close" width="12" height="12" viewBox="0 0 12 12" xmlns="http://www.w3.org/2000/svg">' +
```
Comment (rybmat) [inline extensions/wikia/BannerNotifications/js/BannerNotifications.js:23]: can you add here a TODO comment with the ticket number for supporting ds icons in JS?

File: extensions/wikia/DesignSystem/bower.json [hunk 2]
```diff
@@ -3,6 +3,6 @@
   "private": true,
 
   "dependencies": {
-    "design-system": "Wikia/design-system#8.7.0"
+    "design-system": "Wikia/design-system#XW-3171"
```
Comment (rybmat) [inline extensions/wikia/DesignSystem/bower.json]: remember about release

**Review by rybmat — APPROVED** [review]

File: extensions/wikia/BannerNotifications/js/BannerNotifications.js [hunk 3]
```diff
@@ -14,25 +14,57 @@
...
+			warnIcon: '<svg class="wds-icon wds-icon-small" width="18" height="16" viewBox="0 0 18 16" xmlns="http://www.w3.org/2000/svg"><path d="M17.928 15.156c.1.178.096.392-.013.565a.603.603 0 0 1-.515.28H.6a.607.607 0 0 1-.515-.28.544.544 0 0 1-.013-.564L8.472.278c.21-.37.847-.37 1.056 0l8.4 14.878zM8 5.99v4.02A1 1 0 0 0 9 11c.556 0 1-.444 1-.99V5.99A1 1 0 0 0 9 5c-.556 0-1 .444-1 .99zM8 13c0 .556.448 1 1 1 .556 0 1-.448 1-1 0-.556-.448-1-1-1-.556 0-1 .448-1 1z" fill-rule="evenodd"/></svg>'
+		},
+
+		defaultTimeout = 10000,
 		types = {
-			'notify': 'blue',
-			'confirm': 'green',
-			'error': 'red',
-			'warn': 'yellow'
+			'notify': {
```
Comment (hakubo) [inline extensions/wikia/BannerNotifications/js/BannerNotifications.js]: not needed ''

File: extensions/wikia/BannerNotifications/templates/BannerNotifications.mustache [hunk 4]
```diff
@@ -1,6 +1,12 @@
-<div class="banner-notification">
-	<button class="close wikia-chiclet-button">
-		<img>
-	</button>
-	<div class="msg">{{{content}}}</div>
-</div>
\ No newline at end of file
+<div class="wds-banner-notification {{typeClassName}}">
+	<div class="wds-banner-notification__icon">
+		{{{icon}}}</div>
```
Comment (hakubo) [inline extensions/wikia/BannerNotifications/templates/BannerNotifications.mustache]: maybe that should be on one line (line 2)

File: extensions/wikia/BannerNotifications/templates/BannerNotifications_Confirmation.php [hunk 5]
```diff
@@ -1,10 +1,17 @@
...
+	<div class="wds-banner-notification-container">
+		<? foreach ( $notifications as $notification ): ?>
+			<div class="wds-banner-notification <?= $notification['class'] ?>">
+				<div class="wds-banner-notification__icon">
+					<?= DesignSystemHelper::renderSvg( $notification['icon'],
+						'wds-icon wds-icon-small' ) ?>
+				</div>
+				<span class="wds-banner-notification__text">
+				<?= $notification['message'] ?>
+			</span>
```
Comment (hakubo) [inline extensions/wikia/BannerNotifications/templates/BannerNotifications_Confirmation.php]: weird indenting?

**Review by hakubo — CHANGES_REQUESTED** [review]

**Review by JamesHawking — APPROVED** [review]

**Review by hakubo — APPROVED** [review]

**Pull request merged**